code/modules/mob/mob.dm

```diff
@@ -173,7 +173,7 @@
 	var/range = 7
 	if(vision_distance)
 		range = vision_distance
-	for(var/mob/M in get_hearers_in_view(range, src))
+	for(var/mob/M as() in hearers(range, src))
```

Comment:
```suggestion
	for(var/mob/M as() in hearers(range, T))
```

Reply:
I haven't seen any bugs with visible_message() yet but this will likely prevent them before they crop up.